pkg/jx/cmd/controller_build.go

```diff
@@ -321,6 +328,116 @@ func (o *ControllerBuildOptions) createPromoteStepActivityKey(buildName string,
 	}
 }
 
+// completeBuildSourceInfo sets the PR author and PR title from GitHub in the given PA
+// If the PA is a branch build it then sets the commit author and last commit message
+func (o *ControllerBuildOptions) completeBuildSourceInfo(activity *v1.PipelineActivity) {
+
+	log.Infof("[BuildInfo] Completing build info for PipelineActivity=%s\n", activity.Name)
+
+	gitInfo, err := gits.ParseGitURL(activity.Spec.GitURL)
+	if err != nil {
+		log.Warnf("Cannot parse git repo URL [%s]. Error: %s\n", activity.Spec.GitURL, err)
+		return
```

Comment:
any reason to shallow the error? I would propagate it to the caller and let the caller decide how to handle it.

Reply:
Computing values of this new fields should be a _best effort_ action, I don't want to block any other process because of this (a temporary GH issue, or a rate limit hit should not prevent the local operation of updating an activity). That's why I just logged errors and moved on.

Reply:
I can imaging that this method can return an error, which gets logged from where it is called, without blocking anything.

This style of logging a message and return nothing is not very common in Go.

Reply:
@ccojocar changed. Care to re-review please?